python/ray/serve/controller.py

```diff
@@ -394,6 +404,53 @@ def deploy_group(self, deployment_args_list: List[Dict]) -> List[bool]:
 
         return [self.deploy(**args) for args in deployment_args_list]
 
+    def deploy_app(
+        self,
+        import_path: str,
+        runtime_env: str,
+        deployment_override_options: List[Dict],
+    ) -> None:
+        """Kicks off a task that deploys a Serve application.
+
+        Cancels any previous in-progress task that is deploying a Serve
+        application.
+
+        Args:
+            import_path (str): Serve deployment graph's import path
+            runtime_env (str): runtime_env to run the deployment graph in
+            deployment_override_options (List[Dict]): All dictionaries should
+                contain argument-value options that can be passed directly
+                into a set_options() call. Overrides deployment options set
+                in the graph itself.
+        """
+
+        if self.config_deployment_request_ref is not None:
+            ray.cancel(self.config_deployment_request_ref)
+
+        @ray.remote(runtime_env=runtime_env, max_calls=1)
+        def run_graph(import_path: str, deployment_override_options: List[Dict]):
```

Comment:
can you make this a top level function so it isn't exported every time? 

https://docs.ray.io/en/latest/ray-core/tasks/patterns/redefine-task-actor-loop.html